tests/Models/UserSeasonScoreAggregateTest.php

```diff
@@ -0,0 +1,193 @@
+
+        $this->createRoomWithPlay(15, 'A');
+
+        $userScore->refresh();
+        $this->assertSame($userScore->total_score, (float) 15);
+    }
+
+    public function testAddHigherScoreInParentRoom(): void
+    {
+        $this->createRoomWithPlay(15, 'A');
+
+        $userScore = UserSeasonScoreAggregate::where('user_id', $this->user->getKey())
+            ->where('season_id', $this->season->getKey())
+            ->first();
+
+        $this->assertSame($userScore->total_score, (float) 15);
+
+        $this->createRoomWithPlay(10, 'A');
+
+        $userScore->refresh();
+        $this->assertSame($userScore->total_score, (float) 15);
+    }
+
+    public function testAddSameScoreInChildAndParentRoom(): void
+    {
+        $this->createRoomWithPlay(10, 'A');
+
+        $userScore = UserSeasonScoreAggregate::where('user_id', $this->user->getKey())
+            ->where('season_id', $this->season->getKey())
+            ->first();
+
+        $this->assertSame($userScore->total_score, (float) 10);
+
+        $this->createRoomWithPlay(10, 'A');
+
+        $userScore->refresh();
+        $this->assertSame($userScore->total_score, (float) 10);
+    }
+
+    public function testAddScoreInChildRoomOnly(): void
+    {
+        $this->createRoom('A');
+        $this->createRoomWithPlay(10, 'A');
+
+        $userScore = UserSeasonScoreAggregate::where('user_id', $this->user->getKey())
+            ->where('season_id', $this->season->getKey())
+            ->first();
+
+        $this->assertSame($userScore->total_score, (float) 10);
+    }
+
+    public function testAddScoreInSecondRoomOnly(): void
+    {
+        $this->createRoom('A');
+        $this->createRoomWithPlay(10, 'B');
+
+        $userScore = UserSeasonScoreAggregate::where('user_id', $this->user->getKey())
+            ->where('season_id', $this->season->getKey())
+            ->first();
+
+        $this->assertSame($userScore->total_score, (float) 10);
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->season = Season::factory()->create([
+            'score_factors' => [1, 0.75, 0.5],
+        ]);
+        $this->user = User::factory()->create();
+    }
+
+    private function createRoom(string $groupIndicator): Room
+    {
+        $room = Room::factory()->create([
+            'category' => 'spotlight',
+        ]);
+
+        SeasonRoom::factory()->create([
+            'group_indicator' => $groupIndicator,
+            'room_id' => $room,
+            'season_id' => $this->season,
+        ]);
+
+        return $room;
+    }
+
+    private function createRoomWithPlay(float $totalScore, string $groupIndicator): Room
+    {
+        $room = $this->createRoom($groupIndicator);
+
+        $playlistItem = PlaylistItem::factory()->create([
+            'owner_id' => $room->host,
+            'room_id' => $room,
+        ]);
+
+        $this->roomAddPlay($this->user, $playlistItem, [
```

Comment:
more like `static::roomAddPlay` but either works I guess 🤷

Reply:
UserScoreAggregateTest used ``$this->roomAddPlay()`` so just mimicked it here, but static probably makes more sense indeed 🤔